connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/EmailAggregationProcessor.java

```diff
@@ -0,0 +1,104 @@
+package com.redhat.cloud.notifications.connector.email;
+
+import com.redhat.cloud.notifications.connector.email.config.EmailConnectorConfig;
+import com.redhat.cloud.notifications.connector.email.model.DailyDigestSection;
+import com.redhat.cloud.notifications.connector.email.model.EmailNotification;
+import com.redhat.cloud.notifications.qute.templates.IntegrationType;
+import com.redhat.cloud.notifications.qute.templates.TemplateDefinition;
+import com.redhat.cloud.notifications.qute.templates.TemplateService;
+import io.quarkus.logging.Log;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class EmailAggregationProcessor {
+
+    @Inject
+    TemplateService templateService;
+
+    @Inject
+    EmailConnectorConfig emailConnectorConfig;
+
+    public String aggregate(final EmailNotification emailNotification, final String orgId, String emailTitle) {
+        Map<String, Object> environment = (Map<String, Object>) emailNotification.eventData().get("environment");
+        String bundleName = emailNotification.eventData().get("bundle_name").toString();
+
+        List<Map<String, Object>> listApplicationAggregatedData = (List<Map<String, Object>>) emailNotification.eventData().get("application_aggregated_data_list");
+
+        Log.info("Starting aggregation for bundleName: " + bundleName);
+        Map<String, DailyDigestSection> dataMap  = new HashMap<>();
+        for (Map<String, Object> applicationAggregatedDataAsMap : listApplicationAggregatedData) {
```

Comment:
The code would be easier to read and maintain if we could deserialize into pojos instead of using Maps and unchecked casts in the aggregation process.